src/nvim/os/stdpaths.c

```diff
@@ -0,0 +1,101 @@
+#include <stdbool.h>
+
+#include "nvim/os/stdpaths_defs.h"
+#include "nvim/os/os.h"
+#include "nvim/path.h"
+#include "nvim/memory.h"
+
+/// Names of the environment variables, mapped to XDGVarType values
+static const char *xdg_env_vars[] = {
+  [kXDGConfigHome] = "XDG_CONFIG_HOME",
+  [kXDGDataHome] = "XDG_DATA_HOME",
+  [kXDGCacheHome] = "XDG_CACHE_HOME",
+  [kXDGRuntimeDir] = "XDG_RUNTIME_DIR",
+  [kXDGConfigDirs] = "XDG_CONFIG_DIRS",
+  [kXDGDataDirs] = "XDG_DATA_DIRS",
+};
+
+/// Defaults for XDGVarType values
+///
+/// Used in case environment variables contain nothing. Need to be expanded.
+static const char *const xdg_defaults[] = {
+  // Windows, Apple stuff are just shims right now
+#ifdef WIN32
+  // Windows
```

Comment:
17.10.2015, 21:41, "Felipe Morales" <notifications@github.com>:
> In src/nvim/os/stdpaths.c:
>
>> +static const char *xdg_env_vars[] = { > + [kXDGConfigHome] = "XDG_CONFIG_HOME", > + [kXDGDataHome] = "XDG_DATA_HOME", > + [kXDGCacheHome] = "XDG_CACHE_HOME", > + [kXDGRuntimeDir] = "XDG_RUNTIME_DIR", > + [kXDGConfigDirs] = "XDG_CONFIG_DIRS", > + [kXDGDataDirs] = "XDG_DATA_DIRS", > +}; > + > +/// Defaults for XDGVarType values > +/// > +/// Used in case environment variables contain nothing. Need to be expanded. > +static const char *const xdg_defaults[] = { > + // Windows, Apple stuff are just shims right now > +#ifdef WIN32 > + // Windows
>
> If we leave this empty for now, what would it mean for people who are building/using nvim in Windows already?

That this will fail with syntax error. #endif is wrongly placed.

If it is correctly placed it should SEGV.

> —
> Reply to this email directly or view it on GitHub.